plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx

```diff
@@ -91,8 +115,31 @@ const ProductCollectionContent = (
 		return null;
 	}
 
+	// Let's not render anything until default attributes are set.
+	if ( ! isInitialAttributesSet.current ) {
+		return null;
+	}
+
 	return (
 		<div { ...blockProps }>
+			{ attributes.previewState?.isPreview && (
+				<Button
+					variant="primary"
+					size="small"
+					style={ {
+						position: 'absolute',
+						top: 0,
+						right: 0,
+						zIndex: 1000,
+					} }
+					showTooltip
+					label={ attributes.previewState?.previewMessage }
+					className="wc-block-product-collection__preview-button"
+				>
+					Preview
+				</Button>
+			) }
+
 			<InspectorControls { ...props } />
 			<InspectorAdvancedControls { ...props } />
```

Comment:
Both of these are type errors, because you provided the wrong type for the props.

I think you probably intend not to pass `preview` to either, so their prop type should be

`Omit< ProductCollectionEditComponentProps, 'preview' >`